var/spack/repos/builtin/packages/samrai/package.py

```diff
@@ -53,24 +55,26 @@ class Samrai(Package):
     depends_on("zlib")
     depends_on("hdf5+mpi")
     depends_on("boost")
+    depends_on("m4")
```

Comment:
So I am not so sure `depends_on('m4', type='build')` is the right thing to do here. The problem is the numerical kernels for Samrai AMR patches are typically written in `.m4` files and preprocessed into `.f` Fortran files in a makefile, and if m4 weren't to exist after the build, this would break. Maybe I am still ignorant to the dependency type.